src/App.tsx

```diff
@@ -1,24 +1,243 @@
+    setIsLoading(true);
+    setRequest(true);
+    addTodos(newTodo)
+      .then((createdTodo) => {
+        setTodo((currentTodo) => [...currentTodo, createdTodo]);
+        setTitle('');
+        setIsLoading(false);
+      })
+      .catch(() => {
+        setErrorMessage('Unable to add a todo');
+        setTimeout(() => setErrorMessage(''), 3000);
+      })
+      .finally(() => {
+        setTempTodo(null);
+        setIsLoading(false);
+        setRequest(false);
+      });
+    const temp: Todo = Object.assign(newTodo, { id: 0 });
+
+    setTempTodo(temp);
+  };
+
+  const handleDelete = (todoId: number) => {
+    setLoadingId([todoId]);
+    setIsLoaderActive(true);
+    deleteTodos(todoId)
+      .then(() => {
+        setTodo(prevTodos => prevTodos.filter(todo => todo.id !== todoId));
+        setErrorMessage('');
+      })
+      .catch(() => {
+        setErrorMessage('Unable to delete todo');
+      })
+      .finally(() => {
+        setLoadingId([]);
+        setIsLoading(false);
+        setIsLoaderActive(false);
+      });
+  };
+
+  const handleClearCompleted = () => {
+    const deletePromises = todos
+      .filter(({ completed }) => completed)
+      .map(({ id }) => deleteTodos(id));
+
+    Promise.all(deletePromises)
+      .then(() => {
+        return getTodos(USER_ID);
+      })
+      .then((updatedTodos) => {
+        setTodo(updatedTodos);
+      })
+      .catch(() => {
+        isSomeTodosCompleted = false;
+        setErrorMessage('Unable to delete a todo');
+      });
+  };
+
+  const handleDeleteUpdate = (todo: Todo, newTodoTitle: string) => {
+    updateTodo({
+      id: todo.id,
+      title: newTodoTitle,
+      userId: todo.userId,
+      completed: todo.completed,
+    })
+      .then(updatedTodo => {
+        setTodo(prevState => prevState.map(currentTodo => (
+          currentTodo.id !== updatedTodo.id
+            ? currentTodo
+            : updatedTodo
+        )));
+      })
+      .finally(() => {
+        setLoadingId([]);
+        setIsLoaderActive(false);
+      });
+  };
+
+  const handleToggleTodo = (todo: Todo) => {
+    setLoadingId((prevTodoId) => [...prevTodoId, todo.id]);
+
+    return updateTodo({
+      ...todo,
+      completed: !todo.completed,
+    })
+      .then(updatedTodo => {
+        setTodo(prevState => prevState.map(currentTodo => (
+          currentTodo.id !== updatedTodo.id
+            ? currentTodo
+            : updatedTodo
+        )));
+      })
+      .catch(() => {
+        const errorMessag = 'Unable to toggle a todo';
+
+        setErrorMessage(errorMessag);
+        throw new Error(errorMessage);
+      })
+      .finally(() => {
+        setLoadingId((prevTodoId) => prevTodoId.filter(id => id !== todo.id));
+      });
+  };
+
+  const isAllCompleted = todos.every(todo => todo.completed);
+
+  const handleToggleAll = () => {
+    const activeTodos = todos.filter(todo => !todo.completed);
+
+    const todosForUpdate = isAllCompleted ? todos : activeTodos;
```

Comment:
```suggestion
    const todosForUpdate = isAllCompleted ? todos : todos.filter(todo => !todo.completed);
```
You don't need `activeTodos` if `isAllCompleted`